dumper/app/src/test/java/com/google/edwmigration/dumper/application/dumper/connector/cloudera/manager/ClouderaAPIHostsTaskTest.java

```diff
@@ -0,0 +1,178 @@
+import com.google.edwmigration.dumper.application.dumper.connector.cloudera.manager.ClouderaManagerHandle.ClouderaClusterDTO;
+import com.google.edwmigration.dumper.application.dumper.task.TaskRunContext;
+import java.io.IOException;
+import java.io.Writer;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.util.HashSet;
+import java.util.Set;
+import org.apache.http.HttpEntity;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.util.EntityUtils;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.MockedStatic;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ClouderaAPIHostsTaskTest {
+
+  private ClouderaAPIHostsTask task = new ClouderaAPIHostsTask();
+  private ClouderaManagerHandle handle;
+
+  @Mock private TaskRunContext context;
+  @Mock private ByteSink sink;
+
+  @Mock private Writer writer;
+  @Mock private CharSink charSink;
+  @Mock private CloseableHttpClient httpClient;
+  private URI uri;
+
+  @Before
+  public void setUp() throws Exception {
+    uri = URI.create("http://localhost/api");
+    handle = new ClouderaManagerHandle(uri, httpClient);
+
+    when(sink.asCharSink(eq(StandardCharsets.UTF_8))).thenReturn(charSink);
+    when(charSink.openBufferedStream()).thenReturn(writer);
+  }
+
+  @Test
+  public void clustersExists_do_writes_success() throws Exception {
+    handle.initClusters(
+        ImmutableList.of(
+            ClouderaClusterDTO.create("id1", "first-cluster"),
+            ClouderaClusterDTO.create("id125", "second-cluster")));
+
+    try (MockedStatic<EntityUtils> mockedUtils = mockStatic(EntityUtils.class)) {
+      CloseableHttpResponse responseId1 = mock(CloseableHttpResponse.class);
+      HttpEntity entityId1 = mock(HttpEntity.class);
+      when(responseId1.getEntity()).thenReturn(entityId1);
+
+      CloseableHttpResponse responseCldClst = mock(CloseableHttpResponse.class);
+      HttpEntity entityCldClst = mock(HttpEntity.class);
+      when(responseCldClst.getEntity()).thenReturn(entityCldClst);
+
+      when(httpClient.execute(
+              argThat(
+                  get -> get != null && get.getURI().toString().endsWith("/first-cluster/hosts"))))
+          .thenReturn(responseId1);
+      when(httpClient.execute(
+              argThat(
+                  get -> get != null && get.getURI().toString().endsWith("/second-cluster/hosts"))))
+          .thenReturn(responseCldClst);
+
+      mockedUtils
+          .when(() -> EntityUtils.toString(entityId1))
+          .thenReturn("{'clusterName':'first-cluster'}");
+      mockedUtils
+          .when(() -> EntityUtils.toString(entityCldClst))
+          .thenReturn("{\n'clusterName':'second-cluster'\n\r}");
```

Comment:
Observation: This is fine, but as a general note, I'm always a bit cautious using mocks, because it's never quite clear what one is testing. Out of interest, I did find https://github.com/square/okhttp/tree/master/mockwebserver but I've no experience with it. My previous experience is mostly with Spring's MockMvc, which is brilliant, and doesn't use the network stack. So this observation might be interesting, but might not be useful.